zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/BackupBeanTest.java

```diff
@@ -157,4 +158,117 @@ private boolean containsMBean(Set<ZKMBeanInfo> mbeanSet, String mbeanName, boole
         .findAny();
     return foundMBean.isPresent();
   }
+
+  @Test
+  public void testMBeanUpdate() throws Exception {
+    MockBackupManager backupManager = new MockBackupManager(dataDir, dataDir, dataDir, backupTmpDir, 15,
+        new FileSystemBackupStorage(backupConfig), TEST_NAMESPACE, 0);
+    BackupBean backupBean = backupManager.getBackupBean();
+
+    Assert.assertEquals(0, backupBean.getMinutesSinceLastSuccessfulSnapshotIteration());
+    Assert.assertEquals(0, backupBean.getMinutesSinceLastSuccessfulTxnLogIteration());
+
+    String[] nodeNames = {"/firstNode", "/secondNode", "/thirdNode", "/fourthNode", "/fifthNode"};
+
+    createNode(connection, nodeNames[0]);
+    backupManager.getSnapBackup(snapLog, true).run(1);
+    backupManager.getLogBackup(snapLog, true).run(1);
+    Assert.assertTrue(backupBean.getLastSnapshotIterationDuration() > 0);
+    Assert.assertTrue(backupBean.getLastTxnLogIterationDuration() > 0);
+    Assert.assertEquals(1, backupBean.getNumberOfSnapshotFilesBackedUpLastIteration());
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedTxnLogIterations());
+    Assert.assertFalse(backupBean.getSnapshotBackupActiveStatus());
+    Assert.assertFalse(backupBean.getTxnLogBackupActiveStatus());
+
+    createNode(connection, nodeNames[1]);
+    backupManager.getSnapBackup(snapLog, false).run(1);
+    backupManager.getLogBackup(snapLog, true).run(1);
+    Assert.assertEquals(1, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedTxnLogIterations());
+
+    createNode(connection, nodeNames[2]);
+    backupManager.getSnapBackup(snapLog, true).run(1);
+    backupManager.getLogBackup(snapLog, false).run(1);
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(1, backupBean.getNumConsecutiveFailedTxnLogIterations());
+
+    createNode(connection, nodeNames[3]);
+    backupManager.getSnapBackup(snapLog, false).run(1);
+    backupManager.getLogBackup(snapLog, false).run(1);
+    Assert.assertEquals(1, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(2, backupBean.getNumConsecutiveFailedTxnLogIterations());
+
+    createNode(connection, nodeNames[4]);
+    backupManager.getSnapBackup(snapLog, true).run(1);
+    backupManager.getLogBackup(snapLog, true).run(1);
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedSnapshotIterations());
+    Assert.assertEquals(0, backupBean.getNumConsecutiveFailedTxnLogIterations());
+
+    Thread.sleep(60 * 1000);
```

Comment:
Nit: `L` for longs

Reply:
`backupBean.getNumConsecutiveFailedTxnLogIterations()` in an int typed variable.